docs: add reference to NPM and small how-to (#2)
rico-projects/rico-angular: README.adoc

Please find an example in the https://github.com/rico-projects/rico-samples[rico-samples repository].

== How-To

> Using Rico in your Angular application 

Reviewer: ```suggestion
The following steps describes how you can use Rico in your Angular application.
```

=== Step 1: Add Dependencies

> Add dependencies to rico-js and rico-angular to your project:

Reviewer: ```suggestion
Add dependencies to `rico-js` and `rico-angular` to your project:
```

 npm install git+https://github.com/rico-project/rico-js.git#1.0.0-CR.4
 npm install rico-angular

So that these lines end up in your `package.json`:

    "rico-angular": "^1.0.2",
    "rico-js": "git+https://github.com/rico-project/rico-js.git#1.0.0-CR.4",

=== Step 2: Inject library

Extend the `app.module.ts` by adding the `RicoAngularModule` to the imports:

----
import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';

import { AppComponent } from './app.component';

import { RicoAngularModule } from 'rico-angular';

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule,
    FormsModule,
    RicoAngularModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
----

=== Step 2: Add Code    

Inject RicoService and the https://angular.io/api/core/ApplicationRef[ApplicationRef] into your Component you want to use Rico in:

----
constructor(ricoService: RicoService, appRef: ApplicationRef) {
----

> Connect to Rico Server and create an instance of the controller to get access tothe model you can establish bindings to:

Reviewer: ```suggestion
Connect to a Rico server endpoint and create an instance of the controller to get access to the model for that you can establish bindings to:
```

----

 this.ricoService.connect('http://localhost:8080/todo-app/remoting', this.appRef).then(() => {

      this.ricoService.createController('ToDoController').then((controllerProxy) => {
        AppComponent.LOGGER.info('received proxy after createController:', controllerProxy);
        this.model = controllerProxy.model;
        this.appRef.tick();
        this.controllerProxy = controllerProxy;
      }).catch((error) => {
        AppComponent.LOGGER.error(error);
      });

----

> For a complete working example, please refer to https://github.com/rico-projects/rico-samples/tree/master/todo-list/client-angular[client-angular module] of our https://github.com/rico-projects/rico-samples[ToDo List sample].

Reviewer: ```suggestion
For a complete working example, please refer to the https://github.com/rico-projects/rico-samples/tree/master/todo-list/client-angular[client-angular module] of our https://github.com/rico-projects/rico-samples[ToDo List sample].
```
